File: app/src/main/java/com/nikola/jakshic/dagger/data/local/DotaDatabase.java
package com.nikola.jakshic.dagger.data.local;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;

import com.nikola.jakshic.dagger.model.Competitive;
import com.nikola.jakshic.dagger.model.Hero;
import com.nikola.jakshic.dagger.model.Leaderboard;
import com.nikola.jakshic.dagger.model.Peer;
import com.nikola.jakshic.dagger.model.Player;
import com.nikola.jakshic.dagger.model.SearchHistory;
import com.nikola.jakshic.dagger.model.match.Match;

@Database(
        entities = {
                Player.class,
                SearchHistory.class,
                Competitive.class,
                Match.class,
                Peer.class,
                Hero.class,
                Leaderboard.class},
        version = 1,
        exportSchema = false)

public abstract class DotaDatabase extends RoomDatabase {

    public abstract PlayerDao playerDao();

    public abstract SearchHistoryDao searchHistoryDao();

    public abstract CompetitiveDao competitiveDao();

    public abstract PeerDao peerDao();

    public abstract HeroDao heroDao();

    public abstract MatchDao matchDao();

    public abstract LeaderboardDao leaderboardDao();
}

File: app/src/main/java/com/nikola/jakshic/dagger/data/local/DotaDatabase.kt
package com.nikola.jakshic.dagger.data.local

import android.arch.persistence.room.Database
import android.arch.persistence.room.RoomDatabase
import com.nikola.jakshic.dagger.vo.*
import com.nikola.jakshic.dagger.vo.Match

@Database(entities = arrayOf(
        Competitive::class,
        Leaderboard::class,
        Player::class,
        SearchHistory::class,
        Match::class,
        Hero::class,
        Peer::class,
        Bookmark::class,
        MatchStats::class,
        PlayerStats::class
), version = 5, exportSchema = false)
abstract class DotaDatabase : RoomDatabase() {

    abstract fun competitiveDao(): CompetitiveDao

    abstract fun leaderboardDao(): LeaderboardDao

    abstract fun playerDao(): PlayerDao

    abstract fun searchHistoryDao(): SearchHistoryDao

    abstract fun matchDao(): MatchDao

    abstract fun heroDao(): HeroDao

    abstract fun peerDao(): PeerDao

    abstract fun bookmarkDao(): BookmarkDao

    abstract fun matchStatsDao() : MatchStatsDao

    abstract fun playerStatsDao() : PlayerStatsDao
}